v1/ao/layer.go

```diff
@@ -210,11 +219,16 @@ func (s *layerSpan) BeginSpan(spanName string, args ...interface{}) Span {
 	return s.BeginSpanWithOptions(spanName, SpanOptions{}, args...)
 }
 
+func addTagsFromOpts(opts SpanOptions, tags map[string]interface{}) {
+	if opts.WithBackTrace {
+		tags[KeyBackTrace] = string(debug.Stack())
```

Comment:
Not sure if I read the code correctly, it seems like this will get called either on trace start or `produceKVsFromAllSources` which called for each `BeginSpanWithOptions` (is that for all spans or only some spans).

If it's for all spans, then backtrace is added to all the span and `debug.Stack` is invoked for each of them? :)

Reply:
Yes it's called for all spans but the backtrace is only added when the user set the `WithBackTrace` flag to true explicitly for a span.